activesupport/lib/active_support/cache.rb

```diff
@@ -77,24 +77,22 @@ def lookup_store(*store_option)
       #
       # The +key+ argument can also respond to +cache_key+ or +to_param+.
       def expand_cache_key(key, namespace = nil)
-        expanded_cache_key = (namespace ? "#{namespace}/" : "").dup
-
-        if prefix = ENV["RAILS_CACHE_ID"] || ENV["RAILS_APP_VERSION"]
-          expanded_cache_key << "#{prefix}/"
-        end
-
-        expanded_cache_key << retrieve_cache_key(key)
-        expanded_cache_key
+        [namespace, ENV["RAILS_CACHE_ID"] || ENV["RAILS_APP_VERSION"], retrieve_cache_key(key)].compact.join("/")
```

Comment:
This is more allocations in a reasonably hot path, and also seems less readable to me

Reply:
This is less allocation: in previous impelementation you allocate many strings. In this implementation you allocate one array and the dup of it (via `compact`).
We can change `compact` to `tap(&:compact!)` if you care about allocations.

Put 👎  and I will revert to previous implementation. As I am not super concerned.

I am not sure what is "a reasonably hot path". Can you explain?

Reply:
Previously it was one string in `expanded_cache_key` (two if `namespace`), plus one more if `prefix`. With this change it's the array literal, the compacted array, and the joined string.

"hot path" = code-path that's called very frequently, deserving of performance concern; reasonably (in this context) = not very, but somewhat [hot].